libraries/AP_RTC/AP_RTC.cpp

```diff
@@ -205,6 +205,25 @@ uint32_t AP_RTC::get_time_utc(int32_t hour, int32_t min, int32_t sec, int32_t ms
     return static_cast<uint32_t>(total_delay_ms);
 }
 
+// get date and time.  Returns true on success and fills in year, month, day, hour, min, sec and ms
+// year is the regular Gregorian year, month is 0~11, day is 1~31, hour is 0~23, minute is 0~59, second is 0~60 (1 leap second)
+bool AP_RTC::get_date_and_time_utc(uint16_t& year, uint8_t& month, uint8_t& day, uint8_t &hour, uint8_t &min, uint8_t &sec) const
+{
+    // get local time of day in ms
+    uint64_t time_us = 0;
+    if (!get_utc_usec(time_us)) {
+        return false;
+    }
+    time_t utc_sec = time_us / (1000U * 1000U);
+    struct tm* tm = gmtime(&utc_sec);
```

Comment:
check nullptr

Reply:
Txs again for the review.  nullptr check added and also I confirmed the resulting time matches UTC time exactly.